Assets/MRTK/SDK/Experimental/Drawstring/Scripts/ElasticHandleLogic.cs

```diff
@@ -0,0 +1,123 @@
+    /// Implements a two-handle elastic "stretch" logic, which allows for
+    /// either one or two pointers to stretch along a particular axis.
+    /// 
+    /// Built around the differential equations for a damped harmonic oscillator.
+    /// Intended for use with critically and over-damped oscillators.
+    /// 
+    /// Usage:
+    /// When a manipulation starts, call Setup.
+    /// Call Update() to compute an iteration of the system and obtain the calculated
+    /// distance between the handles.
+    /// </summary>
+    internal class ElasticHandleLogic
+    {
+        private Vector3 leftInitialPosition = Vector3.zero;
+        private Vector3 rightInitialPosition = Vector3.zero;
+
+        private LinearElasticSystem elasticSystem = null;
+
+        private bool isSetup;
+
+        /// <summary>
+        /// Initialize system with source info from controllers/hands.
+        /// Left and right handles must both be specified, but this does not
+        /// preclude single-hand interaction when Update() is called.
+        /// </summary>
+        /// <param name="leftHandleStart">World position of "left" handle point.</param>
+        /// <param name="rightHandleStart">World position of "right" handle point.</param>
+        /// <param name="extentInfo">Properties of the linear 1-D elastic extent to be manipulated</param>
+        /// <param name="elasticProperties">Properties of the elastic material/spring.</param>
+        /// <param name="leftHandleVelocity">Optional, initial velocity in 1-dimensional stretch space</param>
+        /// <param name="rightHandleVelocity">Optional, initial velocity in 1-dimensional stretch space</param>
+        public virtual void Setup(Vector3 leftHandleStart, Vector3 rightHandleStart,
+                                    ElasticExtentProperties<float> extentInfo, ElasticProperties elasticProperties,
+                                    float leftHandleVelocity = 0.0f, float rightHandleVelocity = 0.0f)
+        {
+            isSetup = true;
+            leftInitialPosition = leftHandleStart;
+            rightInitialPosition = rightHandleStart;
+            elasticSystem = new LinearElasticSystem((leftHandleStart - rightHandleStart).magnitude, 0.0f, extentInfo, elasticProperties);
+        }
+
+        /// <summary>
+        /// Update the internal state of the damped harmonic oscillator, given the left and right pointer positions.
+        /// Note, both left and right input positions are nullable; set the pointer position to null
+        /// if only one of the handles is being interacted with. Returns the calculated distance between the
+        /// handles.
+        /// 
+        /// </summary>
+        /// <param name="leftPointer">World position of the pointer manipulating the left handle.</param>
+        /// <param name="rightPointer">World position of the pointer manipulating the right handle.</param>
+        /// <param name="deltaTime">Amount of time to simulate this step.</param>
```

Comment:
Nit: Specify units.

In general when doing anything related to time, always specify units. Otherwise they could be in seconds, 100ns, milliseconds, etc.